### PR TITLE
Fix scalar timestamp

### PR DESCRIPTION
**Pull request by shantanualsi**
**What this PR does / why we need it**:
This PR fixes an issue in literal expressions where the timestamp returned in response was incorrect. The expected time should be the current time.

image

The date returned here is from 2021.
image

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: pkg/querier/queryrange/codec.go [hunk 1]
```diff
@@ -1314,7 +1314,7 @@ func toProtoScalar(v loghttp.Scalar) []queryrangebase.SampleStream {
 	res = append(res, queryrangebase.SampleStream{
 		Samples: []logproto.LegacySample{{
 			Value:       float64(v.Value),
-			TimestampMs: v.Timestamp.UnixNano() / 1e6,
+			TimestampMs: v.Timestamp.Unix(),
```
Comment (jeschkies) [inline pkg/querier/queryrange/codec.go:1317]: Is this returning milliseconds? I'll take a look in a bit

Comment (shantanualsi) [reply pkg/querier/queryrange/codec.go]: i changed the PR to return seconds from `evalLiteral`. ptal.

File: pkg/logql/engine.go [hunk 2]
```diff
@@ -448,7 +448,7 @@ func (q *query) evalLiteral(_ context.Context, expr *syntax.LiteralExpr) (promql
 		return nil, err
 	}
 	s := promql.Scalar{
-		T: q.params.Start().UnixNano() / int64(time.Millisecond),
```
Comment (jeschkies) [inline pkg/logql/engine.go:451]: Checkout line 468. I think that one is wron. And then there's 474 that just uses `UnixMilli()` :thinking:

File: pkg/querier/queryrange/codec.go (hunk 1 above)
Comment (jeschkies) [reply pkg/querier/queryrange/codec.go]: So, `v.Timestamp` is using the Prometheus `model.Time` which is in milliseconds. So this should be `TimestampMs: int64(v.Timestamp)`.

**Pull request closed without merging**